File: js/task3c/S3_PIPE_06.js
const pipeSafe = (...fns) => value => {
  try {
    return {
      ok: true,
      value: fns.reduce((v, fn) => fn(v), value)
    };
  } catch (error) {
    return { ok: false, error };
  }
};

const risky = x => {
  if (x < 0) throw new Error("Negative!");
  return x * 2;
};

console.log(pipeSafe(risky)(5));
console.log(pipeSafe(risky)(-1));

File: py/task3c/S3_PIPE_06.py
def pipe_safe(*fns):
    def run(x):
        try:
            for fn in fns:
                x = fn(x)
            return {"ok": True, "value": x}
        except Exception as e:
            return {"ok": False, "error": str(e)}
    return run


safe = pipe_safe(lambda x: int(x), lambda x: x * 2)
print(safe("5"))
print(safe("x"))
